### PR TITLE
Added custom config.js to node.js page

### PR DESCRIPTION
**Pull request by dradulescu**
I think this is the right place for it, but not 100% sure.

### CODE REVIEW

**Comment by dradulescu** [comment]
@shiroyasha ping :)

File: docs/programming-languages/javascript-and-node-js.md [hunk 1]
```diff
@@ -174,3 +172,40 @@ for details on pre-installed browsers and testing tools on Semaphore.
 [macos-javascript]: https://docs.semaphoreci.com/ci-cd-environment/macos-mojave-xcode-11-image/#javascript-via-node-js
 [docker-env]: https://docs.semaphoreci.com/ci-cd-environment/custom-ci-cd-environment-with-docker/
 [node-docker-image]: https://hub.docker.com/r/semaphoreci/node
+
+## Custom config.js
+
+You can create a custom config.js using [secrets](/guided-tour/environment-variables-and-secrets/#managing-sensitive-data-with-secrets) as files and environment variables.
+
+
+An example configuration file at project_name/config/config.js path might look like this:
```
Comment (shiroyasha) [inline docs/programming-languages/javascript-and-node-js.md:181]: Use code blocks (backticks) when inserting paths into the text.

File: docs/programming-languages/javascript-and-node-js.md [hunk 2]
```diff
@@ -174,3 +172,40 @@ for details on pre-installed browsers and testing tools on Semaphore.
...
+```
```
Comment (shiroyasha) [inline docs/programming-languages/javascript-and-node-js.md:182]: Add language to all code blocks. `javascript` in this case.

File: docs/programming-languages/javascript-and-node-js.md [hunk 3]
```diff
@@ -174,3 +172,40 @@ for details on pre-installed browsers and testing tools on Semaphore.
 [macos-javascript]: https://docs.semaphoreci.com/ci-cd-environment/macos-mojave-xcode-11-image/#javascript-via-node-js
 [docker-env]: https://docs.semaphoreci.com/ci-cd-environment/custom-ci-cd-environment-with-docker/
 [node-docker-image]: https://hub.docker.com/r/semaphoreci/node
+
+## Custom config.js
+
+You can create a custom config.js using [secrets](/guided-tour/environment-variables-and-secrets/#managing-sensitive-data-with-secrets) as files and environment variables.
```
Comment (shiroyasha) [inline docs/programming-languages/javascript-and-node-js.md:178]: Please use prescriptive language in the docs, for guiding the reader.

Change `You can`, `You might` constructs into `Create a `, `Add a`.

File: docs/programming-languages/javascript-and-node-js.md [hunk 4]
```diff
@@ -174,3 +172,40 @@ for details on pre-installed browsers and testing tools on Semaphore.
 [macos-javascript]: https://docs.semaphoreci.com/ci-cd-environment/macos-mojave-xcode-11-image/#javascript-via-node-js
 [docker-env]: https://docs.semaphoreci.com/ci-cd-environment/custom-ci-cd-environment-with-docker/
 [node-docker-image]: https://hub.docker.com/r/semaphoreci/node
+
+## Custom config.js
```
Comment (shiroyasha) [inline docs/programming-languages/javascript-and-node-js.md:176]: This section deals with setting up Database configuration. A better title for this is `Connecting your Node.js application to the test Database`

Comment (shiroyasha) [reply docs/programming-languages/javascript-and-node-js.md:176]: This change might affect the rest of the flow as well.

**Comment by shiroyasha** [comment]
Reviewed it and left a few comments.

File: docs/programming-languages/javascript-and-node-js.md [hunk 5]
```diff
@@ -174,3 +172,40 @@ for details on pre-installed browsers and testing tools on Semaphore.
...
+}
+```
+Create the secrets for `DATABASE_NAME`, `DATABASE_USERNAME` and `DATABASE_PASSWORD`
+while taking care to start postgres through [sem-service](/ci-cd-environment/sem-service-managing-databases-and-services-on-linux/#sem-service-managing-databases-and-services-on-linux) with the appropriate parameters:
+
+``` bash
+sem-service start postgres --username="$DATABASE_USERNAME" --password="$DATABASE_PASSWORD" --db="$DATABASE_NAME"
+```
+
+Then use the credentials in `config.js` to instantiate [Sequelize](https://github.com/sequelize/sequelize):
```
Comment (shiroyasha) [inline docs/programming-languages/javascript-and-node-js.md:200]: Missing `,`. `Then, use the credentials...`.

File: docs/programming-languages/javascript-and-node-js.md [hunk 6]
```diff
@@ -174,3 +172,40 @@ for details on pre-installed browsers and testing tools on Semaphore.
 [macos-javascript]: https://docs.semaphoreci.com/ci-cd-environment/macos-mojave-xcode-11-image/#javascript-via-node-js
 [docker-env]: https://docs.semaphoreci.com/ci-cd-environment/custom-ci-cd-environment-with-docker/
 [node-docker-image]: https://hub.docker.com/r/semaphoreci/node
+
+## Connecting your Node.js application to the test Database
+
+Create a custom `config.js` using [secrets](/guided-tour/environment-variables-and-secrets/#managing-sensitive-data-with-secrets) as files and environment variables.
```
Comment (shiroyasha) [inline docs/programming-languages/javascript-and-node-js.md:178]: Thinking out loud: Why do we need to store this file in secrets? Can the customers store this in the repository? There is nothing sensitive in that file.

File: docs/programming-languages/javascript-and-node-js.md [hunk 7]
```diff
@@ -174,3 +172,37 @@ for details on pre-installed browsers and testing tools on Semaphore.
 [macos-javascript]: https://docs.semaphoreci.com/ci-cd-environment/macos-mojave-xcode-11-image/#javascript-via-node-js
 [docker-env]: https://docs.semaphoreci.com/ci-cd-environment/custom-ci-cd-environment-with-docker/
 [node-docker-image]: https://hub.docker.com/r/semaphoreci/node
+
+## Connecting your Node.js application to the test Database
+
+Create an example configuration file at `project_name/config/config.js` with the following:
```
Comment (shiroyasha) [inline docs/programming-languages/javascript-and-node-js.md:178]: Now that we are talking about his repository, we can say:

> Set up a `config/config.js` file in your repository with the following content:

File: docs/programming-languages/javascript-and-node-js.md [hunk 8]
```diff
@@ -174,3 +172,37 @@ for details on pre-installed browsers and testing tools on Semaphore.
...
+
+module.exports = {
+  postgres: {
+    database: process.env.DATABASE_NAME,
+    username: process.env.DATABASE_USERNAME,
+    password: process.env.DATABASE_PASSWORD
+  },
+}
+```
+Define the [secrets](/guided-tour/environment-variables-and-secrets/#managing-sensitive-data-with-secrets) for `DATABASE_NAME`, `DATABASE_USERNAME` and `DATABASE_PASSWORD`
```
Comment (shiroyasha) [inline docs/programming-languages/javascript-and-node-js.md:190]: This also doesn't have to be a secret. Values for name, username and password are usually not sensitive.

We can say:

> Start a Postgres database in your job with [sem-service](docs):
>
> export DATABASE_USERNAME="test-user"
> export DATABASE_PASSWORD="keyboard-cat"
> export DATABASE_NAME="test-db"
>
> sem-service start postgres --username="$DATABASE_USERNAME" --password="$DATABASE_PASSWORD" --db="$DATABASE_NAME"

**Comment by shiroyasha** [comment]
We had a lot of back-and-forths on this one, but the final docs are 🎖️. Great work!

**Comment by dradulescu** [comment]
Thank you for bearing with me on this! :bow:

**Pull request merged**